titiler/dependencies.py

```diff
@@ -68,12 +69,67 @@ def TMSParams(
     return tms.get(TileMatrixSetId.name)
 
 
+def parse_color(value: Union[List[int], str]) -> List[int]:
+    """Parse RGB/RGBA color and return valid rio-tiler compatible RGBA colormap entry.
+
+    Args:
+        value (str or list of int): HEX encoded or list RGB or RGBA colors.
+
+    Returns:
+        list: RGBA values.
+
+    Examples:
+        >>> parse_color("#FFF")
+        [255, 255, 255, 255]
+
+        >>> parse_color("#FF0000FF")
+        [255, 0, 0, 255]
+
+        >>> parse_color("#FF0000")
+        [255, 0, 0, 255]
+
+        >>> parse_color([255, 255, 255])
+        [255, 255, 255, 255]
+
+    """
+    if isinstance(value, str):
+        hex_pattern = (
+            r"^#"
+            r"(?P<red>[a-fA-F0-9]{1,2})"
+            r"(?P<green>[a-fA-F0-9]{1,2})"
+            r"(?P<blue>[a-fA-F0-9]{1,2})"
+            r"(?P<alpha>[a-fA-F0-9]{1,2})?"
+            r"$"
+        )
+        match = re.match(hex_pattern, value)
+        if match:
+            factor = 1 if len(match.groups()[0]) == 2 else 2
+            value = [
+                int(n * factor, 16) for n in match.groupdict().values() if n is not None
+            ]
+        else:
+            raise Exception(f"Invalid color {value}")
+
+    if len(value) == 3:
+        value += [255]
+
+    return value
+
+
 def ColorMapParams(
-    color_map: ColorMapName = Query(None, description="Colormap name",)
+    colormap_name: ColorMapName = Query(None, description="Colormap name"),
+    colormap: str = Query(None, description="JSON encoded custom Colormap"),
 ) -> Optional[Dict]:
     """Colormap Dependency."""
-    if color_map:
-        return cmap.get(color_map.value)
+    if colormap_name:
+        return cmap.get(colormap_name.value)
+
+    if colormap:
+        return json.loads(
```

Comment:
should we also accept base64 encoded string?

Reply:
I think base64 is less weird than JSON for get requests

Reply:
another way maybe would be to pass a list of value like 
```
colormap=1,#FFFF00&colormap=2,#FF00FF

#equivalent of

colormap={"1": "#FFFF00", "2":"#FF00FF"}
```